tcpassembly/assembly.go

```diff
@@ -256,6 +256,38 @@ func (a *Assembler) FlushOlderThan(t time.Time) (flushed, closed int) {
 	return flushes, closes
 }
 
+// This is much like FlushOlderThan, but doesn't close empty streams...
+// This is useful to get stream moving where don't expect long delays on
+// out-of-order packets, but you do expect long idle sessions or significant
+// time to elapse between a request and response between two correlated streams.
+func (a *Assembler) FlushNoCloseOlderThan(t time.Time) (flushed, closed int) {
```

Comment:
This is a great idea, but not too extensible.  How about the following:

```
type FlushOptions struct {
  T time.Time  // If nonzero, only connections with data older than T are flushed
  CloseAll bool  // If true, ALL connections are closed post flush, not just those that correctly see FIN/RST.
}

func (a *Assembler) Flush() { a.FlushWithOptions(FlushOptions{CloseAll: true}) }
func (a *Assembler) FlushOlderThan(t time.Time) { a.FlushWithOptions(FlushOptions{CloseAll: true, T: t}) }
```

Then, instead of a new FlushNoCloseOlderThan, you could just say `a.FlushWithOptions(FlushOptions{T: t})`